src/polycap-capil.c

```diff
@@ -588,7 +588,7 @@ int polycap_capil_reflect(polycap_photon *photon, polycap_vector3 surface_norm,
 		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_capil_reflect: photon must not be NULL");
 		return -1;
 	}
-	if (&surface_norm == NULL){
+	if (surface_norm == NULL){
 		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_capil_reflect: surface_norm must not be NULL");
```

Comment:
Laat gewoon deze check vallen, het is toch niet mogelijk om NULL door te geven voor surface_norm.